src/nodes/peerdfs.cc

```diff
@@ -27,15 +18,11 @@ using namespace std;
 
 namespace eclipse {
 // Constructor & destructor {{{
-PeerDFS::PeerDFS() : Node() { 
-  Settings& setted = context.settings;
+PeerDFS::PeerDFS (Network* net) : Node () { 
+  network = net;
+  net->attach(this);
 
-  int port       = setted.get<int>("network.ports.internal");
-  size           = setted.get<vec_str>("network.nodes").size();
-  disk_path      = setted.get<string>("path.scratch");
-  nodes          = setted.get<vector<string>>("network.nodes");
-
-  network   = new AsyncNetwork<P2P>(this, port);
+  size = context.settings.get<vec_str>("network.nodes").size();
```

Comment:
It might be confusing. We can use something more specific like, "net_size" or "num_nodes". I'm not sure what would be the best.